core/src/main/java/org/kiwix/kiwixmobile/core/history/viewmodel/State.kt

```diff
@@ -0,0 +1,59 @@
+/*
+ * Kiwix Android
+ * Copyright (c) 2020 Kiwix <android.kiwix.org>
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package org.kiwix.kiwixmobile.core.history.viewmodel
+
+import org.kiwix.kiwixmobile.core.extensions.HeaderizableList
+import org.kiwix.kiwixmobile.core.history.adapter.HistoryListItem
+import org.kiwix.kiwixmobile.core.history.adapter.HistoryListItem.DateItem
+import org.kiwix.kiwixmobile.core.history.adapter.HistoryListItem.HistoryItem
+import java.text.SimpleDateFormat
+import java.util.Locale
+
+data class State(
+  val historyItems: List<HistoryItem>,
+  val showAll: Boolean,
+  val currentZimId: String?,
+  val searchTerm: String = ""
+) {
+  val isInSelectionState = historyItems.any(HistoryItem::isSelected)
+  private val dateFormatter = SimpleDateFormat("d MMM yyyy", Locale.getDefault())
+
+  fun getHistoryListItems(): List<HistoryListItem> =
+    HeaderizableList<HistoryListItem, HistoryItem, DateItem>(historyItems
+      .filter {
+        it.historyTitle.contains(
+          searchTerm,
+          true
+        ) && (it.zimId == currentZimId || showAll)
+      }
+      .sortedByDescending { dateFormatter.parse(it.dateString) })
```

Comment:
put this into the HistoryDao, I am pretty sure this is the exclusive consumer of the history data so we are happy to change it if it makes it more convenient

`Box.asFlowable` accepts a query as its first param, there should be an example in one of the usages.